app/controllers/v1/purls_controller.rb

```diff
@@ -17,8 +17,19 @@ def destroy
       return render json: { error: "already deleted" }, status: :conflict if @purl.deleted?
 
       @purl.mark_deleted
-      UpdatePurlMetadataService.new(@purl).delete!
-      UpdateStacksFilesService.delete!(@purl.cocina_object)
+
+      if Settings.features.versioned_files && VersionedFilesService.versioned_files?(druid: @purl.druid)
```

Comment:
Can this conditional go in the `UpdateStacksFilesService`?

Reply:
I think it's important to encapsulate logic like "where are we storing this stuff", and that encapsulation should protect the controller from needing to know this.  I think it's possible to put this in UpdateStacksFilesService (even if you need to move what is currently there to "LegacyStacksFilesService".

Reply:
OK @jcoyne ; this is done.